server/meca/routes.js

```diff
@@ -0,0 +1,39 @@
+const logger = require('@pubsweet/logger')
+const config = require('config')
+// TODO avoid this recursive dependency
+const ManuscriptManager = require('@elifesciences/xpub-server/entities/manuscript')
+
+module.exports = app => {
+  const apiKey = config.get('meca.apiKey')
+
+  app.post('/meca-result/:id', (req, res) => {
+    const manuscriptId = req.params.id
+    const authHeader = req.get('authorization')
+    const token = authHeader && authHeader.match(/Bearer (.+)/) && RegExp.$1
+
+    if (token !== apiKey) {
+      logger.warn('MECA callback received with invalid API key', {
+        manuscriptId,
+      })
+      res.status(403).send({ error: 'Invalid API key' })
+      return
+    }
+
+    const { body } = req
+    logger.info('MECA callback received', { manuscriptId, body })
+    const status =
+      body.result === 'success'
+        ? ManuscriptManager.statuses.MECA_IMPORT_SUCCEEDED
+        : ManuscriptManager.statuses.MECA_IMPORT_FAILED
+
+    ManuscriptManager.save({ id: manuscriptId, status })
+      .then(() => res.sendStatus(204))
+      .catch(err => {
+        logger.error('Failed to process MECA callback', {
+          manuscriptId,
+          error: err.message,
+        })
+        res.status(500).send({ error: err.message })
+      })
```

Comment:
It may be prudent to declare the catch statement against the `.save` promise. If `res.sendStatus` throws an error, it would get caught and another error may get thrown by `res.status(500)`

Reply:
Not worth addressing